ingest/src/main/scala/hydra.ingest/http/HydraIngestJsonSupport.scala

```diff
@@ -19,15 +19,55 @@ package hydra.ingest.http
 import hydra.core.ingest.IngestionReport
 import hydra.core.marshallers.HydraJsonSupport
 import hydra.core.protocol.IngestorStatus
+import hydra.core.transport.ValidationStrategy
 import hydra.ingest.IngestorInfo
+import hydra.ingest.services.IngestionFlowV2.V2IngestRequest
+import spray.json.JsObject
+
+private object HydraIngestJsonSupport {
+  private final case class IntermediateV2IngestRequest(
+                                                        key: JsObject,
+                                                        value: Option[JsObject],
+                                                        validationStrategy: Option[ValidationStrategy]
+                                                      )
+}
 
-/**
-  * Created by alexsilva on 2/18/16.
-  */
 trait HydraIngestJsonSupport extends HydraJsonSupport {
 
+  import HydraIngestJsonSupport._
   import spray.json._
 
+  implicit object ValidationStrategyFormat extends RootJsonFormat[ValidationStrategy] {
+
+    def read(json: JsValue): ValidationStrategy = json match {
+      case JsString(s) if s.toLowerCase == "strict" => ValidationStrategy.Strict
+      case JsString(s) if s.toLowerCase == "relaxed" => ValidationStrategy.Relaxed
+      case _ =>
+        import scala.reflect.runtime.{universe => ru}
+        val tpe = ru.typeOf[ValidationStrategy]
+        val clazz = tpe.typeSymbol.asClass
+        throw DeserializationException(
+          s"expected a ValidationStrategy of ${clazz.knownDirectSubclasses}, but got $json"
+        )
+    }
+
+    def write(obj: ValidationStrategy): JsValue = {
+      JsString(obj.toString)
+    }
+  }
+
+  private implicit val intermediateV2IngestRequestFormat: JsonFormat[IntermediateV2IngestRequest] =
+    jsonFormat3(IntermediateV2IngestRequest)
+
+  implicit object V2IngestRequestFormat extends RootJsonFormat[V2IngestRequest] {
+    override def read(json: JsValue): V2IngestRequest = {
+      val int = intermediateV2IngestRequestFormat.read(json)
+      V2IngestRequest(int.key.compactPrint, int.value.map(_.compactPrint), int.validationStrategy)
+    }
+
+    override def write(obj: V2IngestRequest): JsValue = ???
```

Comment:
If this is intentional, can you put a comment or something so that we know that it was intentional.

Reply:
We mostly just need to decide what our team's best practice is. The ??? already throws a NotImplementedError. If we're cool with that, great. If we want to make sure it is always handled by our team, maybe a custom error would make more sense discussing the intentional nature of not implementing it.

Reply:
Yeah, I think ??? is fine because it already throws that exception. I added a comment for now just to make it clear that it was on purpose.